core/src/test/scala/unit/kafka/utils/CoreUtilsTest.scala

```diff
@@ -135,7 +135,7 @@ class CoreUtilsTest extends Logging {
     try {
       Await.result(Future.traverse(1 to count) { _ =>
         Future {
-          CoreUtils.atomicGetOrUpdate(map, 0, {
+          map.getOrElseUpdate(0, {
```

Comment:
Should we just remove this test?

Reply:
Hi @mimaison, 
I kept this test to ensure that it still behaves the same as before. 
Maybe we could add a comment noting that this test could be removed in a future release. What do you think?

Reply:
If I recall correctly, the discussion focused on performance rather than behavior correction. If the behavior is inconsistent or incorrect, we should never replace it. So +1 to removing this test.

@frankvicky, could you please share the JMH results? The concern was about potential regressions in read-heavy scenarios

Reply:
Yes, you are right.
I will paste the result later.

Reply:
```
Benchmark                                                            (mapSize)  (writePercentage)  Mode  Cnt  Score   Error  Units
ConcurrentMapBenchmark.testConcurrentHashMapComputeIfAbsentReadOnly        100                0.1  avgt    5  4.182 ± 0.006  ns/op
ConcurrentMapBenchmark.testConcurrentHashMapGetReadOnly                    100                0.1  avgt    5  3.372 ± 0.024  ns/op
JMH benchmarks done
```